applications/ParticleMechanicsApplication/python_scripts/apply_mpm_3d_rotating_dirichlet_condition_process.py

```diff
@@ -0,0 +1,146 @@
+            
+
+            imposed_disp += new_rotation_matrix * (self._TransposeMatrix(self._rotation_matrix) * radius) - radius
+            mpc.SetValuesOnIntegrationPoints(KratosParticle.MPC_IMPOSED_DISPLACEMENT,[imposed_disp],self.model_part.ProcessInfo)
+
+            # Update normal vector
+            normal = mpc.CalculateOnIntegrationPoints(KratosParticle.MPC_NORMAL,self.model_part.ProcessInfo)[0]
+            modified_normal = new_rotation_matrix * (self._TransposeMatrix(self._rotation_matrix) * normal)
+            mpc.SetValuesOnIntegrationPoints(KratosParticle.MPC_NORMAL,[modified_normal],self.model_part.ProcessInfo)
+
+
+        # Copy rotation matrix
+        self._rotation_matrix = new_rotation_matrix
+
+    ### Protected functions
+    def _ComputeCenterRotation(self):
+        auto_rc = KratosMultiphysics.Vector(3)
+        for mpc in self.model_part.Conditions:
+            auto_rc += mpc.CalculateOnIntegrationPoints(KratosParticle.MPC_COORD,self.model_part.ProcessInfo)[0]
+
+        auto_rc = auto_rc / self.model_part.NumberOfConditions()
+        self.rotation_center = auto_rc
+
+    def _InitializeRotationVariables(self):
+        self._quaternion = KratosMultiphysics.Quaternion()
+        self._quaternion = self._quaternion.Identity()
+        
+        self._delta_quaternion = KratosMultiphysics.Quaternion()
+        self._delta_quaternion = self._delta_quaternion.Identity()
+
+        self._rotation_matrix = KratosMultiphysics.Matrix(3,3)
+        self._rotation_matrix.fill_identity()
+
+    def _ComputeDeltaQuaternion(self, rot_velocity):
+        if((not isinstance(rot_velocity, KratosMultiphysics.Vector)) and rot_velocity.Size() != 3):
+            raise Exception("expected input shall be a KratosMultiphysics.Vector object with size 3")
+
+        omega = math.sqrt(rot_velocity[0]*rot_velocity[0] + rot_velocity[1]*rot_velocity[1] + rot_velocity[2]*rot_velocity[2])
+        if omega < 1.e-12:
+            omega = 1.e-12
+
+        dt = self.model_part.ProcessInfo[KratosMultiphysics.DELTA_TIME]
+        
+        self._delta_quaternion = KratosMultiphysics.Quaternion()
+        self._delta_quaternion.W = math.cos(omega * dt / 2.0)
+        self._delta_quaternion.X = math.sin(omega * dt / 2.0) * rot_velocity[0] / omega
+        self._delta_quaternion.Y = math.sin(omega * dt / 2.0) * rot_velocity[1] / omega
+        self._delta_quaternion.Z = math.sin(omega * dt / 2.0) * rot_velocity[2] / omega
+
+    def _TransposeMatrix(self, A):
+        if(not isinstance(A, KratosMultiphysics.Matrix)):
+            raise Exception("expected input shall be a KratosMultiphysics.Matrix object with equal number of row-column (square)")
+
+        result = KratosMultiphysics.Matrix(A.Size2(),A.Size1())
+        for i in range (A.Size1()):
+            for j in range (A.Size2()):
+                result[j,i] = A[i,j]
```

Comment:
there is no `transpose` function?

Reply:
i havn't found this function in add_matrix_to_python.cpp

Reply:
hmmm
Do you know if the c++ version has it?

Reply:
it is in the amtrix_interface.h 
but i am not sure how to use it in python

Reply:
I checked, it is called `trans`, and it is not exposed (see `add_matrix_to_python.cpp`)
@VeronikaSinger can you please add a test to `test_matrix_interface.py` when you have time, then I will implement the exposure.
Its not urgent / not for this PR but I think we can profit from it in the long run

Reply:
I put this to a seperate PR #7283